cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -307,6 +309,65 @@ class SimpleArray
         }
     }
 
+    explicit SimpleArray(small_vector<size_t> const & shape,
+                         small_vector<size_t> const & stride,
+                         std::shared_ptr<buffer_type> const & buffer,
+                         bool is_c_contiguous = true,
+                         bool is_f_contiguous = false)
+        : SimpleArray(buffer)
+    {
+        if (buffer)
+        {
+            if (shape.size() != stride.size())
+            {
+                throw std::runtime_error("SimpleArray: shape and stride size mismatch");
+            }
+
+            if (is_c_contiguous)
+            {
+                if (stride[stride.size() - 1] != 1)
+                {
+                    throw std::runtime_error("SimpleArray: C contiguous stride must end with 1");
+                }
+                for (size_t it = 0; it < shape.size() - 1; ++it)
+                {
+                    if (stride[it] != shape[it + 1] * stride[it + 1])
+                    {
+                        throw std::runtime_error("SimpleArray: C contiguous stride must match shape");
+                    }
+                }
+            }
+            if (is_f_contiguous)
+            {
+                if (stride[0] != 1)
+                {
+                    throw std::runtime_error("SimpleArray: Fortran contiguous stride must start with 1");
+                }
+                for (size_t it = 0; it < shape.size() - 1; ++it)
+                {
+                    if (stride[it + 1] != shape[it] * stride[it])
+                    {
+                        throw std::runtime_error("SimpleArray: Fortran contiguous stride must match shape");
+                    }
+                }
+            }
```

Comment:
Check the C contiguous and F contiguous array.

Reply:
The current implementation is good.

We should refactor it to use a distinct helper (class or function) in a later PR (may use a separate issue to track).

Reply:
By the way I have a questions:
1. How could we pass and receive the `bool is_c_contiguous` and `bool is_f_contiguous`  elegantly? 
I have considered passing the numpy flags directly, but the constructor may not only deal with the numpy array. Is it a good way to design the customed flags by  `enum class types` to deal with every possible input?

Reply:
I don't think the class template `SimpleArray` needs to know f/c continuity.  The sanity check may happen in the Python wrapper which has the information from Numpy.  Standalone helper (static or free function) allows you to check in C++ and pybind11 wrapper.  It could make sense to create a stride object to organize the logics.

It is ok to add more metadata in `SimpleArray`, including f/c continuity, but it is a separate topic.